instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/kafka/utils.py

```diff
@@ -0,0 +1,202 @@
+        return KafkaPropertiesExtractor._extract_argument(
+            "headers", 3, None, args, kwargs
+        )
+
+    @staticmethod
+    def extract_send_partition(args: Any, kwargs: Any):
+        return KafkaPropertiesExtractor._extract_argument(
+            "partition", 4, None, args, kwargs
+        )
+
+    @staticmethod
+    def extract_client_id(client: AIOKafkaClient):
+        return client._client_id
+
+
+ProduceHookT = Optional[Callable[[Span, List, Dict], None]]
+ConsumeHookT = Optional[Callable[[Span, ABCRecord, List, Dict], None]]
+
+
+class KafkaContextGetter(textmap.Getter[textmap.CarrierT]):
+    def get(self, carrier: textmap.CarrierT, key: str) -> Optional[List[str]]:
+        if carrier is None:
+            return None
+
+        for item_key, value in carrier:
+            if item_key == key:
+                if value is not None:
+                    return [value.decode()]
+        return None
+
+    def keys(self, carrier: textmap.CarrierT) -> List[str]:
+        if carrier is None:
+            return []
+        return [key for (key, value) in carrier]
+
+
+class KafkaContextSetter(textmap.Setter[textmap.CarrierT]):
+    def set(self, carrier: textmap.CarrierT, key: str, value: str) -> None:
+        if carrier is None or key is None:
+            return
+
+        if value:
+            value = value.encode()
+        carrier.append((key, value))
+
+
+_kafka_getter = KafkaContextGetter()
+_kafka_setter = KafkaContextSetter()
+
+
+def _enrich_span(
+    span: Span,
+    bootstrap_servers: List[str],
+    topic: str,
+    partition: Optional[int],
+    client_id: str,
+):
+    span.set_attributes(
+        {
+            SpanAttributes.MESSAGING_SYSTEM: "kafka",
+            SpanAttributes.MESSAGING_DESTINATION: topic,
+            SpanAttributes.MESSAGING_URL: json.dumps(bootstrap_servers),
+            SpanAttributes.MESSAGING_DESTINATION_KIND: "topic",
+            SpanAttributes.MESSAGING_KAFKA_CLIENT_ID: client_id,
+        }
+    )
+
+    if span.is_recording():
+        if partition is not None:
+            span.set_attribute(SpanAttributes.MESSAGING_KAFKA_PARTITION, partition)
+
+
+def _get_span_name(operation: str, topic: str):
+    return f"{topic} {operation}"
+
+
+def _wrap_send(tracer: Tracer, produce_hook: ProduceHookT) -> Callable:
+    def _traced_send(func, instance: AIOKafkaProducer, args, kwargs):
+        headers = KafkaPropertiesExtractor.extract_send_headers(args, kwargs)
+        if headers is None:
+            headers = []
+            kwargs["headers"] = headers
+
+        topic = KafkaPropertiesExtractor.extract_send_topic(args, kwargs)
+        bootstrap_servers = KafkaPropertiesExtractor.extract_bootstrap_servers(
+            instance.client
+        )
+        partition = KafkaPropertiesExtractor.extract_send_partition(args, kwargs)
+        client_id = KafkaPropertiesExtractor.extract_client_id(instance.client)
+
+        span_name = _get_span_name("publish", topic)
```

Comment:
I would use `send` here. As `test_utils` is using also `send` and it would be align with python-kafka